### PR TITLE
[NIFI-10349] Add maximum object age property to list s3

### PR DESCRIPTION
**Pull request by marcocharlie**
# Summary

I needed to add the maximum object age property in ListS3 as minimum object age property is already available (https://github.com/apache/nifi/pull/2491).

Signed-off-by: 
Marco [marco.carlino@pagopa.it](mailto:marco.carlino@pagopa.it)
Matteo [matteo.ronci@pagopa.it](mailto:matteo.ronci@pagopa.it) @mronci

[NIFI-10349](https://issues.apache.org/jira/browse/NIFI-10349)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [x] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [x] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java [hunk 1]
```diff
@@ -217,6 +217,14 @@ public class ListS3 extends AbstractS3Processor implements VerifiableProcessor {
             .defaultValue("0 sec")
             .build();
 
+    public static final PropertyDescriptor MAX_AGE = new Builder()
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java:220]: You need an additional validation in `customValidate` to ensure the MAX_AGE is greater than the MIN_AGE.

**Review by MikeThomsen — CHANGES_REQUESTED** [review]
Overall looking good. What changes do you want to make before taking it off of draft?

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java (hunk 1 above)
Comment (marcocharlie) [reply nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java:221]: thank you very much we added `createMaxAgeValidator` and fixed `testListIgnoreByMaxAge`, ready for review

**Comment by marcocharlie** [comment]
@MikeThomsen how does it work to go on with the PR? cc @mronci

**Comment by marcocharlie** [comment]
Hi @MikeThomsen, how can we merge this PR to main branch? Have we missed something? cc @mronci

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java [hunk 2]
```diff
@@ -217,6 +218,15 @@ public class ListS3 extends AbstractS3Processor implements VerifiableProcessor {
...
+            .name("max-age")
+            .displayName("Maximum Object Age")
+            .description("The maximum age that an S3 object must be in order to be considered; any object older than this amount of time (according to last modification date) will be ignored")
```
Comment (turcsanyip) [inline nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java:224]: @marcocharlie I had a look at the code and I think there is a typo here:
```suggestion
            .description("The maximum age that an S3 object can be in order to be considered; any object older than this amount of time (according to last modification date) will be ignored")
```
or `may be in order to be considered`.

**Review by turcsanyip — APPROVED** [review]
@marcocharlie Thanks for the typo fix! It looks good to me now. Also tested it with a simple flow.
Please assign the jira to yourself and also move it to Patch Available (Submit Patch button)!

@MikeThomsen Feel free to merge it unless you have further comments.

+1 from my side

**Comment by marcocharlie** [comment]
thank you @turcsanyip!

**Comment by turcsanyip** [comment]
Merging to main...

@MikeThomsen Feel free to open a follow-up jira if you have any further comments.

**Pull request closed without merging**